packages/react/src/TreeView/TreeView.tsx

```diff
@@ -829,6 +847,19 @@ const TrailingVisual: React.FC<TreeViewVisualProps> = props => {
 
 TrailingVisual.displayName = 'TreeView.TrailingVisual'
 
+// ----------------------------------------------------------------------------
+// TreeView.LeadingAction
+
+const LeadingAction: React.FC<IconButtonProps> = props => {
```

Comment:
Can we make the `LeadingAction` accept children similar to `leadingVisual` and `trailingVisual`? This lets us (consumer) pass in any component (and associated props to that component) to the LeadingAction

Reply:
Since we already have a drag trigger component that is accessible and works around some quirks of NVDA, we'd love to use that (instead of rewriting the logic again within `LeadingAction` button) if possible.

Reply:
> We need an icon button slot before the expand/collapse toggle so we can place the drag handle there.

Think I made a small mistake here when asking for an icon button slot. I got it a little mixed up but ideally, we'd want similar functionality to `LeadingVisual` i.e a slot to pass child components

Reply:
https://github.com/primer/react/compare/ayy-bc/addTreeViewDragHandle...ayy-bc/suggestions

Reply:
Merged those changes to this PR

Reply:
See also Eric's comment about tabIndex=-1: https://github.com/primer/react/pull/4546#issuecomment-2107683449

Reply:
Yeah, the drag trigger component that we use has a wrapper around the `IconButton` component

https://github.com/github/github/blob/master/ui/packages/drag-and-drop/components/SortableListTrigger.tsx

Reply:
And I assume its the same as using the leading/trailing visual so the consumer (us) can pass the `tabindex=-1` like we do when consuming TreeView in NestedListView.